compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -238,12 +238,18 @@ pub fn check_crate(tcx: TyCtxt<'_>) {
                 check::maybe_check_static_with_link_section(tcx, item_def_id);
             }
             DefKind::Const if !tcx.generics_of(item_def_id).own_requires_monomorphization() => {
```

Comment:
I would expect `!own_requires_monomorphization` means there are no generics to begin with? Or what does that function mean? It is sadly undocumented...

It does not make sense to me that we'd ever evaluate a *generic* free const item. How would we even pick the instance for the generic parameters...?

Reply:
Ah, there was https://github.com/rust-lang/rust/pull/136429 which I already forgot about. IMO the obvious fix is to just revert that PR...

... but I guess that wouldn't even be enough since even non-generic consts can have where clauses. Right. Uh, we may have to just skip all of those?